### ISSUE TITLE
Multiplayer Campaign - Bots Do Not Persist

### ISSUE THREAD

**Issue by TankerD18** [Regalis11/Barotrauma#3555]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
Bots, bought or initially spawned with, do not reappear upon load of a saved multiplayer campaign.

**Steps To Reproduce**
Start multiplayer campaign with 4 max players, 4 bots.
Hire an additional bot on top of whatever the campaign is started with.
Proceed to conduct a mission arriving at the next station.
Exit the game and shutdown server.
Start new server and load campaign.

**Version**
0.10.4.0

**Additional information**
Happened with a private multiplayer game I played with my buddy where we got one mission in and came back to it the next day to find our bots were gone. Was able to replicate by performing the above steps with no additional players, first load had bots, second load did not. Not running any mods or custom subs.

**Comment by 3e849f2e5c** [comment]
Possibly fixed with https://github.com/Regalis11/Barotrauma/issues/3346

**Comment by DaimonTrilogy** [comment]
Yesterday my friend and I had still the same issue present.
We are playing on the newest steam version.
The Error is replicatable like in [#3346](https://github.com/Regalis11/Barotrauma/issues/3346).

Maybe the Fix is not present in the steam version yet?

**Comment by Regalis11** [comment]
@DaimonTrilogy You're right, the fix is not included in the public steam version yet, so I'm afraid you'll have to wait for the next patch. In the meantime, there are a couple of ways you can work around the issue: https://trello.com/c/2ALYmI30/310-workaround-included-newly-hired-bots-and-newly-bought-subs-and-submarine-upgrades-may-disappear-in-campaign

Sorry for the inconvenience!

**Comment by S1lenos** [comment]
@Regalis11 The solution you recommend tackles an entirely different issue, with Bots not being present after recruiting them.

I had the same issue with disappearing bots in multiplayer. Playing this weekend (August 22nd), we had 2 players and some starting bots + recruited bots, all in all about 7 or 8. IIRC we were able to save the campaign and correctly load it once. However, next time we saved the game and loaded the game, all bots were gone. We reloaded the game again, and also tried undocking and docking, hoping they would return, but they were gone for good.
